spec/models/manageiq/providers/amazon/cloud_manager/metrics_capture_spec.rb

```diff
@@ -58,18 +58,26 @@
   end
 
   context 'counters present' do
-    let(:vm_name) { 'i-091369dd8a8d470ab' }
     let(:ems) { FactoryBot.create(:ems_amazon_with_vcr_authentication, :provider_region => 'eu-central-1') }
+    let(:data) { vm.perf_collect_metrics('realtime').last.first.last }
 
-    subject do
-      with_vcr_data { vm.perf_collect_metrics('realtime').last.first.last.first.last }
+    shared_examples 'available' do
+      it { is_expected.to have_key('mem_usage_absolute_average') }
+      it { is_expected.to have_key('mem_swapped_absolute_average') }
     end
 
-    it { is_expected.to have_key('mem_usage_absolute_average') }
-    it { is_expected.to have_key('mem_swapped_absolute_average') }
+    context 'from linux old way' do
+      subject { with_vcr_data { data.first.last } }
```

Comment:
So this ends up being `vm.perf_collect_metrics('realtime').last.first.last.first.last`?

Reply:
Exactly. Need to see values from one of the data points in the timestamped series.

Reply:
>> So this ends up being vm.perf_collect_metrics('realtime').last.first.last.first.last?
>
> Exactly.

That's even more confusing :smile:

So assuming we start with something like above where we have a `let(:metrics)` then here we'd have
```
>> timestamps = metrics[vm.ems_ref].keys
=> [timestamp1, timestamp2]
>> counters = metrics[vm.ems_ref][timestamps.first].keys
=> [counter1, counter2]
>> value = metrics[vm.ems_ref][timestamps.first][counters.first]
```

Giving the intermediate values names makes it easier for the reader to see what is going on.

Reply:
Or you could do what you did [down there](https://github.com/ManageIQ/manageiq-providers-amazon/pull/522/files#diff-ffbdff4705cd8ab74bed2c08ee3d8d9cR75) and just use the timestamp value since its from a vcr..

Reply:
changed there: https://github.com/ManageIQ/manageiq-providers-amazon/pull/525